spec/requests/conversion_hosts_spec.rb

```diff
@@ -76,44 +82,70 @@
+      results = response.parsed_body["results"]
+
+      expect(results).to match_array([
+        a_hash_including("message" => "Enabling resource id:#{vm.id} type:#{vm.class}", "task_id" => a_kind_of(String)),
+        a_hash_including("message" => "Enabling resource id:#{host.id} type:#{host.class}", "task_id" => a_kind_of(String)),
+      ])
     end
+  end
 
-    it "supports multiple conversion host creation" do
-      api_basic_authorize(collection_action_identifier(:conversion_hosts, :create))
+  context "disable" do
+    let(:zone) { FactoryBot.create(:zone, :name => "api_zone") }
+    let(:ems) { FactoryBot.create(:ems_openstack, :zone => zone) }
+    let(:vm) { FactoryBot.create(:vm_openstack, :ems_id => ems.id) }
+    let(:conversion_host) { FactoryBot.create(:conversion_host, :resource => vm) }
+    let(:conversion_host_url) { api_conversion_host_url(nil, conversion_host) }
 
-      conversion_hosts = [sample_conversion_host_from_vm, sample_conversion_host_from_host]
-      post(api_conversion_hosts_url, :params => gen_request(:create, conversion_hosts))
+    before do
+      allow(conversion_host).to receive(:install_conversion_host_module).and_return(true)
+      allow(conversion_host).to receive(:resource_search).and_return(vm)
+      allow(conversion_host).to receive(:ansible_playbook).and_return({})
+    end
+
+    it "can disable a resource via POST" do
+      api_basic_authorize(action_identifier(:conversion_hosts, :disable, :resource_actions))
+      allow(conversion_host).to receive(:check_conversion_host_role).and_return('disabled')
+
+      post(conversion_host_url, :params => {"action" => "disable"})
 
       expect(response).to have_http_status(:ok)
-      expect_result_resources_to_include_keys("results", expected_attributes)
 
-      results = response.parsed_body["results"]
+      results = response.parsed_body
 
-      expect(ConversionHost.exists?(results.first["id"])).to be_truthy
-      expect(ConversionHost.exists?(results.last["id"])).to be_truthy
-      expect(results).to match_array([a_hash_including("resource_id" => vm.id.to_s), a_hash_including("resource_id" => host.id.to_s)])
+      expect(results['success']).to be_truthy
+      expect(results['href']).to eql("http://www.example.com/api/conversion_hosts/#{conversion_host.id}")
+      expect(results['message']).to eql("Disabling ConversionHost id:#{conversion_host.id} name:#{conversion_host.name}")
+      expect(results['task_id']).to match(/\d+/)
+      expect(MiqTask.exists?(results['task_id'].to_i)).to be_truthy
     end
   end
 
   context "delete" do
-    let(:conversion_host)             { FactoryBot.create(:conversion_host, :resource => FactoryBot.create(:vm)) }
+    let(:zone)                        { FactoryBot.create(:zone, :name => "api_zone") }
+    let(:ems)                         { FactoryBot.create(:ems_openstack, :zone => zone) }
+    let(:vm)                          { FactoryBot.create(:vm_openstack, :ems_id => ems.id) }
```

Comment:
Prefer `:ems => ems` rather than setting id's

Reply:
Tried it, doesn't work, you have to specify `ext_management_system => ems` if you want to do it that way, so the current approach is actually the shortest. I had wondered about that myself when I first worked on these. It probably *should* work, though.

Reply:
Sorry, I meant `:ext_management_system => ems`

Reply:
Updated.